src/pytorch_lightning/strategies/colossalai.py

```diff
@@ -0,0 +1,336 @@
+            def on_load_checkpoint(self, checkpoint) -> None:
+                if not hasattr(self, 'model'):
+                    self.configure_sharded_model()
+        trainer = Trainer(..., accelerator="gpu", precision=16, strategy="colossalai")
+
+    Args:
+        use_chunk: Whether to use chunk-based memory management.
+            It can speed up training, but slightly more memory will be used.
+
+        chunk_size: The size of a chunk.
+            It will be ignored when ``use_chunk=False``.
+            If it's None, a best chunk size will be searched out based on ``chunk_search_range``,
+            ``chunk_search_n_grids`` and ``min_chunk_size``.
+
+        enable_distributed_storage: Whether to storage model in a distributed manner.
+            It reduces memory from 1 to 1/N, but it may slow down training.
+
+        placement_policy: It can be "cpu", "cuda" and "auto".
+
+            * If it's "cpu", parameters, gradients and optimizer states will be offloaded to CPU,
+                which means min CUDA memory will be used.
+            * If it's "cuda", they won't be offloaded, which means max CUDA memory will be used. It's the fastest.
+            * If it's "auto", they are moving dynamically based on CPU and CUDA memory usage.
+                It will utilize heterogeneous memory space evenly and well.
+                Note that "auto" policy can only work well when no other processes use CUDA during your training.
+
+        force_outputs_fp32: Whether to cast outputs to fp32.
+
+        gpu_margin_mem_ratio: The ratio of GPU remaining memory (after the first forward-backward)
+            which will be used by optimizer.
+            This argument will be ignored when ``placement_policy`` is not "auto".
+
+        chunk_search_range: The range of chunk size to search.
+            The actual search range will be from
+            ``max(min_chunk_size, max_param_size)`` to ``max(min_chunk_size, max_param_size) + chunk_search_range``.
+
+        chunk_search_n_grids: The number of intervals in the search range.
+
+        min_chunk_size: The minimum size for a chunk.
+
+        initial_scale: The initial dynamic loss scale value.
+
+        min_scale: The minimum dynamic loss scaling value.
+
+        growth_factor: The multiplication factor for increasing loss scale.
+
+        backoff_factor: The multiplication factor for decreasing loss scale.
+
+        growth_interval: The number of steps to increase loss scale when no overflow occurs.
+
+        hysteresis: The number of overflows before decreasing loss scale.
+
+        max_scale: The maximum dynamic loss scaling value.
+
+    .. _colossalai.nn.optimizer.CPUAdam:
+        https://colossalai.readthedocs.io/en/latest/colossalai/colossalai.nn.optimizer.cpu_adam.html
+
+    .. _colossalai.nn.optimizer.HybridAdam:
+        https://colossalai.readthedocs.io/en/latest/colossalai/colossalai.nn.optimizer.hybrid_adam.html
+
+        strategy_name = "colossalai"
```

Comment:
```suggestion
```
you need to add it outside the doctrines. It's a class variable

Reply:
Done.